Sidekiq Scheduler

cquiro opened the pull request:
## Summary

- Add sidekiq-scheduler gem to project
- Add send_rent_due_email method to RentsMailer
- Create RentDueNotifier worker
- Schedule RentDueNotifier to run every day at 9:00 am GMT-3

## Trello Card

https://trello.com/c/c0W6FnP9/41-notificar-al-usuario-cuando-un-rent-se-vence

## Screenshots

<img width="1389" alt="screen shot 2018-10-03 at 5 04 53 pm" src="https://user-images.githubusercontent.com/12885969/46436120-821a8800-c72e-11e8-80c0-b82160866a28.png">

<img width="1387" alt="screen shot 2018-10-03 at 5 05 04 pm" src="https://user-images.githubusercontent.com/12885969/46436127-8646a580-c72e-11e8-8b16-5cd98b5182f3.png">

cquiro committed:
Add line 'config.active_job.queue_adapter = :test' to config/environments/test.rb

cquiro committed:
Add sidekiq-scheduler gem to project

cquiro committed:
Add send_rent_due_email method to RentsMailer

Add the necessary code so that an email can be sent
to the user advising them to return a rental book on
the day the rental is due.

The send_rent_due_email method was added to the
existing RentsMailer class.

cquiro committed:
Update send_new_rent_email view's locales

cquiro committed:
Create RentDueNotifier worker

RentDueNotifier worker sends emails to the users
whose rentals are due the day the worker is called
using RentsMailer#send_rent_due_email.

cquiro committed:
Schedule RentDueNotifier in config/sidekiq.yml

RentDueNotifier worker was scheduled to run
every day at 9:00 am GMT-3.

cquiro committed:
Fix rubocop offenses

facucompe commented on app/workers/rent_due_notifier.rb:
```diff
@@ -0,0 +1,15 @@
+  include Sidekiq::Worker
+
+  def perform
+    Rent.where(id: rent_ids).find_each do |rent|
```
Why are you doing a where here? You already do it in `rent_ids` method

cquiro replied:
@facucompe My bad! At some point during development I was returning rent_ids and using them in perform, then I realized it was not necessary and removed the `.ids` part in the rent_ids method, but forgot to change the rest.

facucompe commented on config/sidekiq.yml:
```diff
@@ -0,0 +1,3 @@
+:schedule:
+  rent_due_notifier:
+    cron: '0 9 * * * Etc/GMT-3'
```
You can use `'0 9 * * * America/Santiago'`, it's more clear

cquiro committed:
Change Etc/GMT-3 to America/Santiago

The change was made for clarity.

cquiro committed:
Update code in RentDueNotifier

During development some changes where made before commiting and I forgot
to remove some code that was no longer necessary. In this commit that
code is removed.

lmasello commented on app/mailers/rents_mailer.rb:
```diff
@@ -1,15 +1,26 @@
-    @rent = Rent.find(rent_id)
-    @user = @rent.user
-    @book = @rent.book
+    get_rental_data(rent_id)
```
Avoid using getters and setters in the names :) (https://github.com/rubocop-hq/ruby-style-guide#accessor_mutator_method_names)

lmasello reviewed (commented):
Just a comment

cquiro committed:
Replace method names that start with set or get

Replace method names that start with set or get in RentsMailer as per
rubocop-hq/ruby-style-guide.
